src/components/forms/ComboBox/ComboBox.tsx

```diff
@@ -0,0 +1,284 @@
+          </option>
+        ))}
+      </select>
+      <input
+        data-testid="combo-box-input"
+        aria-owns={listID}
+        aria-autocomplete="list"
+        aria-describedby={assistiveHintID}
+        aria-expanded={isOpen}
+        aria-controls=""
+        id={selectID}
+        className="usa-combo-box__input"
+        type="text"
+        role="combobox"
+        ref={inputRef}
+        value={inputValue}
+        onBlur={handleInputBlur}
+        onClick={handleInputClick}
+        onKeyDown={handleInputKeyDown}
+        onChange={(e): void => setInputValue(e.target.value)}
+        // these do not type check:
+        // autocapitalize="off"
+        // autocomplete="off"
+      />
+      <span className="usa-combo-box__clear-input__wrapper" tabIndex={-1}>
+        <button
+          type="button"
+          className="usa-combo-box__clear-input"
+          aria-label="Clear the select contents"
+          onClick={handleClearClick}>
+          &nbsp;
+        </button>
+      </span>
+      <span className="usa-combo-box__input-button-separator">&nbsp;</span>
+      <span className="usa-combo-box__toggle-list__wrapper" tabIndex={-1}>
+        <button
+          type="button"
+          tabIndex={-1}
+          className="usa-combo-box__toggle-list"
+          aria-label="Toggle the dropdown list"
+          onClick={handleArrowClick}>
+          &nbsp;
+        </button>
+      </span>
+      <ul
+        data-testid="combo-box-option-list"
+        tabIndex={-1}
+        id={listID}
+        className="usa-combo-box__list"
+        role="listbox"
+        hidden={!isOpen}>
+        {filteredOptions.map((option, index) => {
+          const focused = option.value === focusedValue
+          const selected = option.value === selectedValue
+          const itemClasses = classnames('usa-combo-box__list-option', {
+            'usa-combo-box__list-option--focused': focused,
+            'usa-combo-box__list-option--selected': selected,
+          })
+
+          return (
+            <li
+              ref={focused ? itemRef : null}
+              value={option.value}
+              key={option.value}
+              className={itemClasses}
+              tabIndex={focused ? 0 : -1}
+              role="option"
+              aria-selected={selected}
+              aria-setsize={64}
+              aria-posinset={index + 1}
+              id={listID + `--option-${index}`}
+              onKeyDown={handleListItemKeyDown}
+              onMouseMove={(): void => handleListItemMouseMove(option)}
+              onClick={(): void => {
+                handleListItemClick(option)
+              }}>
+              {option.label || option.value}
+            </li>
+          )
+        })}
+      </ul>
+
+      <div className="usa-combo-box__status usa-sr-only" role="status"></div>
+      <span id={assistiveHintID} className="usa-sr-only">
+        When autocomplete results are available use up and down arrows to review
```

Comment:
We will need to surface this as a prop so our assistive text can be i18n ready.  We can set a default value of English if needed as per #124

Reply:
I made this a string prop (`assistiveHint`) since I don't think the message is dependent on any state (in the character count component we use a function and pass in the current value). Do you think that is OK, or should we always make this kind of thing a function prop for consistency?

Reply:
That works!  I don't think we should be using a function prop like that too often (hopefully) -  validations might be the main case.